common/build/eslint-plugin-fluid/PUBLISH_GUIDE.md

```diff
@@ -0,0 +1,86 @@
+# Publishing Guide for `@fluidframework/eslint-config-fluid` & `@fluid-internal/eslint-plugin-fluid`
```

Comment:
Nit: What do you think about putting this in a file called "DEV.md"? I've been trying to get us to normalize around a consistent pattern for internal-facing package-level docs like these. These contents could be a second-level heading under a root `@fluidframework/eslint-config-fluid` heading.